### DISCUSSION TITLE
I need help with the control signals.

### DISCUSSION THREAD

**Discussion by juliordzcer** [USC-ACTLab/crazyswarm#666] [Q&A] [answered]
Hello, I am currently a master's student, and I am working with crazyflies, I'm trying to apply controllers by sliding modes, I'm basing myself on your program, but I have a problem about where you send the control signals to the motors, I would greatly appreciate your help aid.

Best regards!

**Comment by jpreiss** [comment, marked as answer]
If you want a controller that directly sets motor voltages, you should probably implement it in the firmware. See [this doc](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/sensor-to-control/) for an overview of the main flight control loop in the firmware. This would not be specific to Crazyswarm. You can get help in the [Bitcraze forums](https://forum.bitcraze.io/viewforum.php?f=6&sid=b7f3c38093afe95ae06177e293b567b7).
